### PR TITLE
Careplan renewal QAs

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/careplan.rb [hunk 1]
```diff
@@ -232,6 +232,13 @@ def rn_just_approved?
       ncm_approval? && rn_approved_on.present? && rn_approved_on_changed?
     end
 
+    def renewed_for_cp2?
+      # Only PCTPs from the 12 months before CP2 can be renewed
+      return false if provider_signed_on.blank? || provider_signed_on < Health::PatientReferral::CP_2_REFERRAL_DATE - 12.months
```
Comment (eanders) [inline app/models/health/careplan.rb:237]: What happens if the provider_signed_on date is >= health::PatientReferral::CP_2_REFERRAL_DATE ?

File: db/health/migrate/20230504143929_renewed_cp1_careplans.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class RenewedCp1Careplans < ActiveRecord::Migration[6.1]
+  def up
+    cp2_date = (Health::PatientReferral::CP_2_REFERRAL_DATE ..)
+
+    # Only careplans touched after the CP2 date can have been renewed...
+    careplans = Health::Careplan.where(updated_at: cp2_date ..).
```
Comment (eanders) [inline db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: cp2_date is already a range, this is weird.  I think you meant:
```suggestion
    careplans = Health::Careplan.where(updated_at: cp2_date).
```

File: app/models/health/careplan.rb (hunk 1 above)
Comment (awisema) [reply app/models/health/careplan.rb:237]: PCP signatures ended on that date -- so, any date after that is not a completed CP1 referral.

File: db/health/migrate/20230504143929_renewed_cp1_careplans.rb (hunk 2 above)
Comment (eanders) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: Just to confirm the logic
The care plan must have been:
1. Updated after the particular date
2. Signed with the year prior to the date
3. approved after the date.

#2 doesn't feel right to me.

Comment (awisema) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: Oops -- I decided I needed the date not the range, and didn't fix cp2_date. Thanks!

Comment (eanders) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: Why does the PCTP need to be updated and approved _after_ the signature?

Comment (awisema) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: The logic is supposed to be:
1. The careplan was updated after CP2 started (so, they have started renewing it)
2. The careplan was signed before CP2 started, but in the last year, (so it was a completed CP1 careplan)
3. The careplan was approved, which means the logic that will handle renewals after deployment would not be triggered, so create the missing QA.

Comment (eanders) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: Ok, that seems to track

Comment (awisema) [reply db/health/migrate/20230504143929_renewed_cp1_careplans.rb:6]: MH is requiring them to renew any current CP1 careplans. This is basically, a mechanical process of having the CC and RN look over it, and send it to the PCP.

**Review by eanders — APPROVED** [review]
Re-targeted `release-74` since `73` is on production, `74` should be going up later this week or early next week.

**Review by eanders — APPROVED** [review]

**Pull request merged**